### PR TITLE
feat(ci): implement mock-ACPI workflow

### PR DESCRIPTION
**Pull request by vprashar2929**
This commit introduces a workflow for testing ACPI functionality using Equinix self-hosted runners. The workflow deploys Kepler using mock-acpi compose setup and runs validator to ensure functionality.

Key-features:
- Workflow is triggered on pull requests that include a specific commit message `/test-acpi`.
- Environment setup is handled by ansible.

### CODE REVIEW

**Comment by vprashar2929** [comment]
Sample CI run: https://github.com/sustainable-computing-io/kepler/actions/runs/10454746478

**Comment by vimalk78** [comment]
LGTM
just one comment, the output of validator run is hard to parse MAPE and MSE values. 
https://github.com/sustainable-computing-io/kepler/actions/runs/10454746478/job/28948217427#step:4:137
```
changed: [localhost] => {"changed": true, "cmd": "cd /opt/kepler/e2e/tools/validator\nhatch run validator -f ./validator.mock-acpi.yaml validate-acpi -d 1m\n", "delta": "0:00:17.526185", "end": "2024-08-19 14:05:32.875316", "failed_when_result": false, "msg": "", "rc": 0, "start": "2024-08-19 14:05:15.349131", "stderr": "Creating environment: default\nInstalling Python distribution: 3.11\nInstalling project in development mode\nChecking dependencies\nSyncing dependencies", "stderr_lines": ["Creating environment: default", "Installing Python distribution: 3.11", "Installing project in development mode", "Checking dependencies", "Syncing dependencies"], "stdout": "  * Generating build and node info ...\n\n  * Build Info\n    - kepler_exporter_build_info{arch=\"amd64\", branch=\"pr-1701\", instance=\"kepler-dev:8888\", job=\"dev\", os=\"linux\", revision=\"d4462319d30d28aa197ba2aa3ed883731ed5bdce\", version=\"v0.7.11-100-gd4462319\"}\n    - kepler_exporter_build_info{arch=\"amd64\", branch=\"main\", instance=\"kepler-latest:8888\", job=\"latest\", os=\"linux\", revision=\"21c7b260903a0ed169a46da64db806ddfa982ba0\", version=\"v0.7.11-101-g21c7b260\"}\n\n  * Node Info\n    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-dev:8888\", job=\"dev\", platform_power_source=\"acpi\", source=\"os\"}\n    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-latest:8888\", job=\"latest\", platform_power_source=\"acpi\", source=\"os\"}\n\n  * Generating spec report ...\n  * Generating validation report\n   - started at: 2024-08-19 14:04:32.760798+00:00\n   - ended   at: 2024-08-19 14:05:32.758796+00:00\n   - duration  : 0:00:59.997998\n\nRun validations\nmock - node\n  - actual  :  sum( rate( kepler_node_platform_joules_total{ job=\"dev\" }[20s] ) ) \n  - expected:  mock_acpi_power1_average/10^6 \n\t MSE : 0.00\n\t MAPE: 0.00 %\n\n  * Generating validate acpi report file and dir\nReport Generated \n * report: /tmp/validator-v0.7.11-100-gd4462319/report-v0.7.11-100-gd4462319.md \n * time-range:   2024-08-19 14:04:32.760798+00:00 - 2024-08-19 14:05:32.758796+00:00", "stdout_lines": ["  * Generating build and node info ...", "", "  * Build Info", "    - kepler_exporter_build_info{arch=\"amd64\", branch=\"pr-1701\", instance=\"kepler-dev:8888\", job=\"dev\", os=\"linux\", revision=\"d4462319d30d28aa197ba2aa3ed883731ed5bdce\", version=\"v0.7.11-100-gd4462319\"}", "    - kepler_exporter_build_info{arch=\"amd64\", branch=\"main\", instance=\"kepler-latest:8888\", job=\"latest\", os=\"linux\", revision=\"21c7b260903a0ed169a46da64db806ddfa982ba0\", version=\"v0.7.11-101-g21c7b260\"}", "", "  * Node Info", "    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-dev:8888\", job=\"dev\", platform_power_source=\"acpi\", source=\"os\"}", "    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-latest:8888\", job=\"latest\", platform_power_source=\"acpi\", source=\"os\"}", "", "  * Generating spec report ...", "  * Generating validation report", "   - started at: 2024-08-19 14:04:32.760798+00:00", "   - ended   at: 2024-08-19 14:05:32.758796+00:00", "   - duration  : 0:00:59.997998", "", "Run validations", "mock - node", "  - actual  :  sum( rate( kepler_node_platform_joules_total{ job=\"dev\" }[20s] ) ) ", "  - expected:  mock_acpi_power1_average/10^6 ", "\t MSE : 0.00", "\t MAPE: 0.00 %", "", "  * Generating validate acpi report file and dir", "Report Generated ", " * report: /tmp/validator-v0.7.11-100-gd4462319/report-v0.7.11-100-gd4462319.md ", " * time-range:   2024-08-19 14:04:32.760798+00:00 - 2024-08-19 14:05:32.758796+00:00"]}
```

**Review by vimalk78 — APPROVED** [review]

**Review by sthaha — APPROVED** [review]

**Comment by vprashar2929** [comment]
> LGTM just one comment, the output of validator run is hard to parse MAPE and MSE values. https://github.com/sustainable-computing-io/kepler/actions/runs/10454746478/job/28948217427#step:4:137
> 
> ```
> changed: [localhost] => {"changed": true, "cmd": "cd /opt/kepler/e2e/tools/validator\nhatch run validator -f ./validator.mock-acpi.yaml validate-acpi -d 1m\n", "delta": "0:00:17.526185", "end": "2024-08-19 14:05:32.875316", "failed_when_result": false, "msg": "", "rc": 0, "start": "2024-08-19 14:05:15.349131", "stderr": "Creating environment: default\nInstalling Python distribution: 3.11\nInstalling project in development mode\nChecking dependencies\nSyncing dependencies", "stderr_lines": ["Creating environment: default", "Installing Python distribution: 3.11", "Installing project in development mode", "Checking dependencies", "Syncing dependencies"], "stdout": "  * Generating build and node info ...\n\n  * Build Info\n    - kepler_exporter_build_info{arch=\"amd64\", branch=\"pr-1701\", instance=\"kepler-dev:8888\", job=\"dev\", os=\"linux\", revision=\"d4462319d30d28aa197ba2aa3ed883731ed5bdce\", version=\"v0.7.11-100-gd4462319\"}\n    - kepler_exporter_build_info{arch=\"amd64\", branch=\"main\", instance=\"kepler-latest:8888\", job=\"latest\", os=\"linux\", revision=\"21c7b260903a0ed169a46da64db806ddfa982ba0\", version=\"v0.7.11-101-g21c7b260\"}\n\n  * Node Info\n    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-dev:8888\", job=\"dev\", platform_power_source=\"acpi\", source=\"os\"}\n    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-latest:8888\", job=\"latest\", platform_power_source=\"acpi\", source=\"os\"}\n\n  * Generating spec report ...\n  * Generating validation report\n   - started at: 2024-08-19 14:04:32.760798+00:00\n   - ended   at: 2024-08-19 14:05:32.758796+00:00\n   - duration  : 0:00:59.997998\n\nRun validations\nmock - node\n  - actual  :  sum( rate( kepler_node_platform_joules_total{ job=\"dev\" }[20s] ) ) \n  - expected:  mock_acpi_power1_average/10^6 \n\t MSE : 0.00\n\t MAPE: 0.00 %\n\n  * Generating validate acpi report file and dir\nReport Generated \n * report: /tmp/validator-v0.7.11-100-gd4462319/report-v0.7.11-100-gd4462319.md \n * time-range:   2024-08-19 14:04:32.760798+00:00 - 2024-08-19 14:05:32.758796+00:00", "stdout_lines": ["  * Generating build and node info ...", "", "  * Build Info", "    - kepler_exporter_build_info{arch=\"amd64\", branch=\"pr-1701\", instance=\"kepler-dev:8888\", job=\"dev\", os=\"linux\", revision=\"d4462319d30d28aa197ba2aa3ed883731ed5bdce\", version=\"v0.7.11-100-gd4462319\"}", "    - kepler_exporter_build_info{arch=\"amd64\", branch=\"main\", instance=\"kepler-latest:8888\", job=\"latest\", os=\"linux\", revision=\"21c7b260903a0ed169a46da64db806ddfa982ba0\", version=\"v0.7.11-101-g21c7b260\"}", "", "  * Node Info", "    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-dev:8888\", job=\"dev\", platform_power_source=\"acpi\", source=\"os\"}", "    - kepler_node_info{components_power_source=\"rapl-sysfs\", cpu_architecture=\"Coffee Lake\", instance=\"kepler-latest:8888\", job=\"latest\", platform_power_source=\"acpi\", source=\"os\"}", "", "  * Generating spec report ...", "  * Generating validation report", "   - started at: 2024-08-19 14:04:32.760798+00:00", "   - ended   at: 2024-08-19 14:05:32.758796+00:00", "   - duration  : 0:00:59.997998", "", "Run validations", "mock - node", "  - actual  :  sum( rate( kepler_node_platform_joules_total{ job=\"dev\" }[20s] ) ) ", "  - expected:  mock_acpi_power1_average/10^6 ", "\t MSE : 0.00", "\t MAPE: 0.00 %", "", "  * Generating validate acpi report file and dir", "Report Generated ", " * report: /tmp/validator-v0.7.11-100-gd4462319/report-v0.7.11-100-gd4462319.md ", " * time-range:   2024-08-19 14:04:32.760798+00:00 - 2024-08-19 14:05:32.758796+00:00"]}
> ```

right I will fix this in a follow up PR
